src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/WebJira.kt

```diff
@@ -9,7 +9,7 @@ import java.net.URLEncoder
 data class WebJira(
     val driver: WebDriver,
     val base: URI,
-    private val adminPassword: String
+    val adminPassword: String
```

Comment:
Adding a new property to the data class is an API breaking change. Please see https://hello.atlassian.net/wiki/spaces/JIRASERVER/blog/2018/12/18/370711047/Is+Kotlin+a+good+choice+for+a+Java+library#Kotlin-data-classes-are-a-poor-choice-for-API.

Reply:
Publishing and existing field does not break ABI.
If you decompile the current `WebJira.kt` to Java, you'll see:
```
   @NotNull
   public final WebDriver component1() {
      return this.driver;
   }

   @NotNull
   public final URI component2() {
      return this.base;
   }

   private final String component3() {
      return this.adminPassword;
   }
```

```
   @NotNull
   public final WebJira copy(@NotNull WebDriver driver, @NotNull URI base, @NotNull String adminPassword) {
      Intrinsics.checkParameterIsNotNull(driver, "driver");
      Intrinsics.checkParameterIsNotNull(base, "base");
      Intrinsics.checkParameterIsNotNull(adminPassword, "adminPassword");
      return new WebJira(driver, base, adminPassword);
   }
```

```
   // $FF: synthetic method
   @NotNull
   public static WebJira copy$default(WebJira var0, WebDriver var1, URI var2, String var3, int var4, Object var5) {
      if ((var4 & 1) != 0) {
         var1 = var0.driver;
      }

      if ((var4 & 2) != 0) {
         var2 = var0.base;
      }

      if ((var4 & 4) != 0) {
         var3 = var0.adminPassword;
      }

      return var0.copy(var1, var2, var3);
   }
```

The private field is already handled.